orchestrator-api/src/main/java/com/sequenceiq/cloudbreak/orchestrator/metadata/OrchestratorMetadata.java

```diff
@@ -0,0 +1,43 @@
+package com.sequenceiq.cloudbreak.orchestrator.metadata;
+
+import java.util.List;
+import java.util.Set;
+
+import com.sequenceiq.cloudbreak.common.orchestration.Node;
+import com.sequenceiq.cloudbreak.common.orchestration.OrchestratorAware;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.orchestrator.state.ExitCriteriaModel;
+
+public class OrchestratorMetadata {
+
+    private final OrchestratorAware stack;
+
+    private final List<GatewayConfig> gatewayConfigs;
+
+    private final Set<Node> nodes;
```

Comment:
since all nodes is accessible with `stack.getAllNodes()`, I would add only an `Optional<Predicate>` filter if filtering needed, what do you think? (like the one you create, `OrcehstratorMetadataFilter`)